parsl/addresses.py

```diff
@@ -9,36 +9,52 @@
 logger = logging.getLogger(__name__)
 
 
-def address_by_route():
+def address_by_route() -> str:
+    """Finds an address for the local host by querying the local routing table
+       for the route to Google DNS.
+
+       This will return an incorrect value when the internet-facing address is
+       not reachable from workers.
+    """
     logger.debug("Finding address by querying local routing table")
     addr = os.popen("/sbin/ip route get 8.8.8.8 | awk '{print $NF;exit}'").read().strip()
     logger.debug("Address found: {}".format(addr))
     return addr
 
 
-def address_by_query():
+def address_by_query() -> str:
+    """Finds an address for the local host by querying ipify. This will
+       return an incorrect value when the host is behind NAT, or when the
+       internet-facing address is not reachable from workers.
+    """
     logger.debug("Finding address by querying remote service")
     addr = requests.get('https://api.ipify.org').text
     logger.debug("Address found: {}".format(addr))
     return addr
 
 
-def address_by_hostname():
+def address_by_hostname() -> str:
+    """Returns the hostname of the local host.
+
+       This will return an incorrect value when the hostname cannot be
+       resolved correctly from workers.
```

Comment:
It is returning a correct value, but a value that may not be resolvable on the workers.